posts/2020-11-20-JNDI-gRPC.adoc

```diff
@@ -0,0 +1,377 @@
+
+        // grab user, address, port params
+        startService(address, port);
+        HelloRequest person = HelloRequest.newBuilder().setName(user).build();
+        HelloReply greeting = greetingService.sayHello(person);
+
+        // send the greeting in a response
+        stopService();
+        }	
+    }
+}
+----
+
+As with the service example, the application must provide the link:https://github.com/grpc/grpc-java/blob/master/examples/src/main/proto/helloworld.proto[helloworld protof definition] along with the protobuf compiler output. All required gRPC client libraries are provided by the `grpcClient-1.0` feature.
+
+[#jndi]
+=== Support for Custom JNDI Names for Enterprise Beans
+
+[#support]
+==== What is Support for Custom JNDI Names for Enterprise Beans?
+
+Support for Custom JNDI Names for Enterprise Beans (EJBs) is an enhancement to all existing enterprise beans features that allows an application to configure and use custom JDNI names for looking up and/or injecting enterprise beans, and to also use legacy default JNDI names from WebSphere traditional.
+
+Prior to this enhancement, Liberty only supported looking up enterprise beans using the specification define JNDI names : `java:global/<app>/<module>/<bean>!<interface>` & variations for `java:app` and `java:module`.
+
+Without any additional configuration, the default JNDI names that were available on WebSphere traditional are now available for applications to use to lookup and/or inject enterprise beans. Also, rather than using the defaults, a custom name for each EJB may be specified in the `ibm-ejb-jar-bnd.xml` file (or `ibm-ejb-jar-bnd.xmi` file for EJB 2.x and 1.x modules). This file is the same format as used by WebSphere traditional. These new JNDI name options are available in addition to the existing specification required names.
+
+[#migration]
+==== Easier Migration for Enterprise Bean Applications
+
+Custom JNDI name support for enterprise beans provides an easier migration path for applications from other platforms (including WebSphere traditional).
+
+Prior to Java EE 6, the Enterprise Beans specification did not prescribe the JNDI names required for enterprise beans, so every platform provided platform specific default names and custom binding file formats. Since Liberty only supported the specification defined JNDI names, migrating applications from other platforms often requires code changes to modify the platform specific JNDI names to the newer specification defined names.
+
+Since Liberty now supports the same default JNDI names, and same binding file format as WebSphere traditional, code changes to modify the JNDI lookup names are no longer required when migrating EJB applications from WebSphere traditional to Liberty.
+Migration from other platforms is also simplified because applications may be migrated without changing code, but instead migrating the other platform specific binding files to the new Liberty platform specific binding file format.
+
```

Comment:
Change to :

Prior to Java EE 6, the Enterprise Beans specification did not prescribe the JNDI names required for enterprise beans, so every platform provided platform specific default names and custom binding file formats. Since Liberty only supported the specification defined JNDI names, migrating applications from other platforms often requires code changes to modify the platform specific JNDI names to the newer specification defined names. Now, migration from other platforms is simplified because applications may be migrated without changing code, but instead migrating the other platform specific binding files to the new Liberty platform specific binding file format. In some cases, use of the new legacy default names may also allow applications to migrate to liberty without specifying custom JNDI names in a binding file.

Reply:
Done